File: ui/src/api/ride.ts
// src/api/ride.ts
import request from '../utils/request'

// Ride 创建请求 DTO
export interface CreateRideDto {
  passengerId: string;
  originAddress: string;
  destAddress: string;
  originLng: number;
  originLat: number;
  destLng: number;
  destLat: number;
}

// Ride 响应
export interface Ride {
  rideId: string;
  driverId: string;
  passengerId: string;
  startTime: string;
  endTime: string;
  origin: string;
  destination: string;
  originLng: number;
  originLat: number;
  destLng: number;
  destLat: number;
  status: string;
  fare?: number;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

// 创建行程
export function createRide(data: CreateRideDto) {
  return request.post<ApiResponse<Ride>>("/api/ride/create", data);
}

// 获取历史行程
export function getHistoryRidesByPassengerId(passengerId: string) {
  return request.get<ApiResponse<Ride[]>>(`/api/ride/history/${passengerId}`);
}

// 更新行程状态
export function updateRideStatus(rideId: string, status: string) {
  return request.post<ApiResponse<Ride>>("/api/ride/change-status", null, {
    params: { rideId, status }
  });
}

// 获取司机附近的行程
export function getNearbyRides(driverId: string) {
  return request.get<ApiResponse<Ride[]>>(`/api/ride/nearby/${driverId}`);
}

// 获取行程详情
export function getRideDetail(rideId: string) {
  return request.get<ApiResponse<Ride>>(`/api/ride/detail/${rideId}`);
}

// 司机接单
export function acceptRide(driverId: string, rideId: string) {
  return request.post<ApiResponse<null>>(`/api/ride/accept`, { driverId, rideId });
}

// 司机到达乘客位置
export function driverArrived(rideId: string) {
  return request.post<ApiResponse<null>>(`/api/ride/arrived/${rideId}`);
}

// 开始行程
export function startRide(rideId: string) {
  return request.post<ApiResponse<null>>(`/api/ride/start/${rideId}`);
}

// 结束行程
export function endRide(rideId: string) {
  return request.post<ApiResponse<null>>(`/api/ride/end/${rideId}`);
}

File: services/service-ride/src/main/java/com/fth/ride/domain/model/Ride.java
package com.fth.ride.domain.model;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_ride")
public class Ride {

    // ===== 基础信息 =====
    private String rideId;
    private String passengerId;
    private String driverId;

    // ===== 位置描述 =====
    private double originLat;
    private double originLng;
    private String originAddress;
    private double destLat;
    private double destLng;
    private String destAddress;

    // ===== 多阶段时间 =====
    private LocalDateTime requestTime; // 乘客发起请求时间
    private LocalDateTime acceptTime;  // 司机接单时间
    private LocalDateTime startTime;   // 乘客上车，开始计费
    private LocalDateTime endTime;     // 行程结束时间

    // ===== 状态 =====
    private String status;

    // ===== 计费/统计数据 =====
    private double distance;  // 行程里程（km）
    private double duration;  // 行程时长（分钟）
    private double price;     // 最终价格

    // ===== 取消原因（用于记录取消状态） =====
    private String cancelReason;

    public Ride(String rideId,
                String passengerId,
                double originLat,
                double originLng,
                double destLat,
                double destLng) {
        this.rideId = rideId;
        this.passengerId = passengerId;
        this.originLat = originLat;
        this.originLng = originLng;
        this.destLat = destLat;
        this.destLng = destLng;
        this.status = "REQUESTED";
        this.requestTime = LocalDateTime.now();
    }

    public Ride() {}

}